kong/tools/public.lua

```diff
@@ -1,26 +1,71 @@
 local pcall = pcall
 local ngx_log = ngx.log
 local ERR = ngx.ERR
+local str_find = string.find
+
+
+local cjson = require "cjson.safe"
```

Comment:
Since no other function in this module is using `cjson` and `string.fin`, maybe we should move those inside the `do ... end` block as well, just like `multipart`.